docs/day-2-operations/Backups.md

```diff
@@ -0,0 +1,272 @@
+   ```
+   oc rollout latest dc/backend-redis
+   ```
+
+5. Rename the `dump.rb` file
+   ```
+   oc rsh $(oc get pods -l 'deploymentConfig=backend-redis' -o json | jq '.items[0].metadata.name' -r) bash -c 'mv ${HOME}/data/dump.rdb ${HOME}/data/dump.rdb-old'
+   ```
+
+6. Rename the `appendonly.aof` file
+   ```
+   oc rsh $(oc get pods -l 'deploymentConfig=system-redis' -o json | jq '.items[0].metadata.name' -r) bash -c 'mv ${HOME}/data/appendonly.aof ${HOME}/data/appendonly.aof-old'
+   ```
+
+7. Move the Backup file to the POD
+
+   ```
+   oc cp ./backend-redis-dump.rdb $(oc get pods -l 'deploymentConfig=backend-redis' -o json | jq '.items[0].metadata.name' -r):/var/lib/redis/data/dump.rdb
+   ```
+
+8. Re-deploy `system-redis` to load the backup
+   ```
+   oc rollout latest dc/backend-redis
+   ```
+
+9. Edit the `redis-config` configmap
+   ```
+   oc edit configmap redis-config
+   ```
+
+10. Uncomment `SAVE` comands in the `redis-config` configmap
+   ```
+    save 900 1
+    save 300 10
+    save 60 10000
+   ```
+
+11. Set `appendonly` to yes in the `redis-config` configmap
+   ```
+   appendonly yes
+   ```
+
+12. Re-deploy `backend-redis` to reload the default configurations
+   ```
+   oc rollout latest dc/backend-redis
+   ```
+
+### `system-redis`
+
+1. Edit the `redis-config` configmap
+   ```
+   oc edit configmap redis-config
+   ```
+
+2. Comment `SAVE` comands in the `redis-config` configmap
+   ```
+    #save 900 1
+    #save 300 10
+    #save 60 10000
+   ```
+
+3. Set `appendonly` to no in the `redis-config` configmap
+   ```
+   appendonly no
+   ```
+
+4. Re-deploy `system-redis` to load the new configurations
+   ```
+   oc rollout latest dc/system-redis
+   ```
+
+5. Rename the `dump.rb` file
+   ```
+   oc rsh $(oc get pods -l 'deploymentConfig=system-redis' -o json | jq '.items[0].metadata.name' -r) bash -c 'mv ${HOME}/data/dump.rdb ${HOME}/data/dump.rdb-old'
+   ```
+
+6. Rename the `appendonly.aof` file
+   ```
+   oc rsh $(oc get pods -l 'deploymentConfig=system-redis' -o json | jq '.items[0].metadata.name' -r) bash -c 'mv ${HOME}/data/appendonly.aof ${HOME}/data/appendonly.aof-old'
+   ```
+
+7. Move the Backup file to the POD
+
+   ```
+   oc cp ./system-redis-dump.rdb $(oc get pods -l 'deploymentConfig=system-redis' -o json | jq '.items[0].metadata.name' -r):/var/lib/redis/data/dump.rdb
+   ```
+
+8. Re-deploy `system-redis` to load the backup
+   ```
+   oc rollout latest dc/system-redis
+   ```
+
+9. Edit the `redis-config` configmap
+   ```
+   oc edit configmap redis-config
+   ```
+
+10. Uncomment `SAVE` comands in the `redis-config` configmap
+   ```
+    save 900 1
+    save 300 10
+    save 60 10000
+   ```
+
+11. Set `appendonly` to yes in the `redis-config` configmap
+   ```
+   appendonly yes
+   ```
+
+12. Re-deploy `system-redis` to reload the default configurations
+   ```
+   oc rollout latest dc/system-redis
+   ```
+
+## Open Issues
+* what about System services and sphinx (index)?
```

Comment:
@andrewdavidmackenzie I don't understand, what is the problem with this? AFAIK sphinx will regenerate the indexes from MySQL as needed, right?

Reply:
Yes sphinx will regenerate indices from `system-database`
No need to backup sphinx indices

Reply:
But will we need some procedure to re-trigger indexing, after a restore of the DB contents?

For system services, we will need to do anything special on restore? Will depend on the documented procedure for restoring the redis with the job queues that system-services executes?

Reply:
Deploy of system-sphinx pod will do it automatically
https://github.com/3scale/3scale-amp-openshift-templates/blob/a5699103fe5a23818ddca52a18c9d862fabbb7d8/amp/amp.yml#L2903

Reply:
ok @hallelujah so in that sense need to document that in case of system DB Restore we would need to redeploy `system-sphinx` to regenerate the indexes right?